pkg/module/sign.go

```diff
@@ -1,5 +1,137 @@
 package module
 
-func Sign() error {
+import (
+	"context"
+	"fmt"
+	cdv2 "github.com/gardener/component-spec/bindings-go/apis/v2"
+	cdv2Sign "github.com/gardener/component-spec/bindings-go/apis/v2/signatures"
+	"github.com/gardener/component-spec/bindings-go/ctf"
+	cdoci "github.com/gardener/component-spec/bindings-go/oci"
+	"github.com/kyma-project/cli/pkg/module/oci"
+	"github.com/kyma-project/cli/pkg/module/signatures"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+type ComponentSignConfig struct {
+	Name           string // Name of the module (mandatory)
+	Version        string // Version of the module (mandatory)
+	RegistryURL    string // Registry URL where unsigned Component descriptor located (mandatory)
+	PrivateKeyPath string // The private key used for signing (mandatory)
+}
+
+func Sign(cfg *ComponentSignConfig, privateKeyPath string, signatureName string, remote *Remote, log *zap.SugaredLogger) ([]*cdv2.ComponentDescriptor, error) {
```

Comment:
Would it not make sense to use the field `PrivateKeyPath` from `ComponentSignConfig` instead of passing an extra parameter?

Additionally, would it then also make sense to move `signatureName` into the config struct as well?

Reply:
good catch!  unfinished refactoring